src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateContractPartyCsvServiceImpl.java

```diff
@@ -57,16 +67,137 @@ protected List<JaggaerContractAddressUploadDto> findVendorAddresses(List<Jaggaer
+        String csvFileName = generateCsvOutputFileName(processingMode);
+        String fullyQualifiedCreationDirectoryFileName = fullyQualifyFileNameToCreationDirectory(csvFileName);
+        LOG.info("generateCsvFile: fullyQualifiedOutputFile = " + fullyQualifiedCreationDirectoryFileName);
+        
+        List<String[]> csvData = buildDataToPrint(jaggaerUploadDtos);
+
+        try (FileWriter fileWriter = new FileWriter(fullyQualifiedCreationDirectoryFileName);
+                CSVWriter writer = new CSVWriter(fileWriter);) {
+            writer.writeAll(csvData);
+        } catch (IOException e) {
+            LOG.error("generateCsvFile, Error writing to file " + fullyQualifiedCreationDirectoryFileName, e);
+            throw new RuntimeException(e);
+        }
+
+    }
+    
+    private String generateCsvOutputFileName(JaggaerContractUploadProcessingMode processingMode) {
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMdd_HHmmss", Locale.US);
+        StringBuilder filename = new StringBuilder(processingMode.csvFileName);
+        filename.append("_").append(sdf.format(dateTimeService.getCurrentDate())).append(CUKFSConstants.FILE_EXTENSIONS.CSV_FILE_EXTENSION);
+        return filename.toString();
+    }
+    
+    private String fullyQualifyFileNameToCreationDirectory(String csvFileName) {
+        StringBuilder filename = new StringBuilder(jaggaerUploadCreationDriectory);
+        filename.append(csvFileName);
+        return filename.toString();
+    }
+    
+    private List<String[]> buildDataToPrint(List<JaggaerContractUploadBaseDto> jaggaerUploadDtos) {
+        List<String[]> csvDataRows = new ArrayList<>();
+        addHeaderRows(csvDataRows);
         
         for (JaggaerContractUploadBaseDto dto : jaggaerUploadDtos) {
-            LOG.info("generateCsvFile, dto: " + dto.toString());
+            if (dto instanceof JaggaerContractPartyUploadDto) {
+                JaggaerContractPartyUploadDto vendorDto = (JaggaerContractPartyUploadDto) dto;
+                csvDataRows.add(builderVendorCSVRowArray(vendorDto));
+            } else if (dto instanceof JaggaerContractAddressUploadDto) {
+                JaggaerContractAddressUploadDto addressDto = (JaggaerContractAddressUploadDto) dto;
+                csvDataRows.add(builderAddressCSVRowArray(addressDto));
+            } else {
+                throw new IllegalArgumentException("Unexpected DTO class found: " + dto.getClass());
+            }
         }
+        return csvDataRows;
+    }
+    
+    protected void addHeaderRows(List<String[]> csvDataRows) {
+        csvDataRows.add(buildPartyHeader());
+        csvDataRows.add(buildAddressHeader());
+        csvDataRows.add(buildContactHeader());
+    }
+    
+    protected String[] buildPartyHeader() {
+        String[] record = { "PARTY", "OverrideDupError", "ERPNumber", "SciQuestID", "ContractPartyName", "DoingBusinessAs",
```

Comment:
Would it be better to put these header labels in a constants class or enum for convenience? The same goes for the other types of headers.

Reply:
OK, moved to an enum